### PR TITLE
[fix]: [CDS-83637]: Adding support for skip validation for Inline pipelines if setting is enabled

### PR DESCRIPTION
**Pull request by rchaudharyy**
Adding support for skip validation for Inline pipelines if setting is enabled

We have done the testing in devspace.

Steps used for testing:
  1. added one user
  2. created dev and prod environment and created one resource group with only dev env.
  3. assigned the created resource group to the created user.
  4. run the pipeline (inline) with two stage one with dev and another with prod env

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)

### CODE REVIEW

File: 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingIdentifiers.java [hunk 1]
```diff
@@ -57,4 +57,6 @@ public class SettingIdentifiers {
       "pagerduty_notification_integration_keys_allowlist";
   public static String AIDA = "aida";
   public static final String ENABLE_STEADY_STATE_FOR_JOBS_KEY_IDENTIFIER = "native_helm_enable_steady_state_for_jobs";
+
```
Comment (ankittiwari-harness) [inline 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingIdentifiers.java:60]: nit: do we need blank line in b/, run clang-format

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java [hunk 2]
```diff
@@ -1032,4 +1029,19 @@ private String readFile(String filename) {
       throw new InvalidRequestException("Could not read resource file: " + filename);
     }
   }
+
+  @Test
+  @Owner(developers = RISHIKESH)
```
Comment (ankittiwari-harness) [inline pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java:1034]: add test for case when setting is disabled.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 5 files at r1, 3 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)* status: 5 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml` line 381 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-Njprxlr6lw4EaWYOnY4:-Njprxlr6lw4EaWYOnY5:bba08dp) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml#L381)):*
> ```YAML
>       - ORGANIZATION
>   - identifier: skip_validation_for_inline_pipelines
>     name: Skip Validation for Inline Pipelines
> ```

Skip Fail-Fast Validation Checks for Pipeline Execute



_Code quote:_
```YAML
Validation
```
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 458 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpsCpbBTm_Q6lbP8eq:-NjpsCpbBTm_Q6lbP8er:by7zhno) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L458)):*
> ```Java
>   }
>   public boolean checkIsSkipValidationForInlinePipelinesEnabled(PipelineEntity pipelineEntity) {
>     String isSkipValidationForInlinePipelinesEnabled = "";
> ```

Keep it false



_Code quote:_
```Java
""
```
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 468 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpsI4v6McP_GsnnCvO:-NjpsI4v6McP_GsnnCvP:b-oigif1) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L468)):*
> ```Java
>       log.error("Failed to fetch setting value for {}", SKIP_VALIDATION_FOR_INLINE_PIPELINES, ex);
>     }
>     return SKIP_VALIDATION_FOR_INLINE_PIPELINES_TRUE_VALUE.equals(isSkipValidationForInlinePipelinesEnabled);
> ```

Convert string to Boolean and compare with Boolean.True.equals



_Code quote:_
```Java
SKIP_VALIDATION_FOR_INLINE_PIPELINES_TRUE_VALUE.equals(isSkipValidationForInlinePipelinesEnabled
```
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 495 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpslZx5hTpVeJgKpEE:-NjpslZx5hTpVeJgKpEF:b-mqlkb7) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L495)):*
> ```Java
>     if (pipelineEntity.getStoreType() == null
>         || (pipelineEntity.getStoreType() == StoreType.INLINE
>             && !checkIsSkipValidationForInlinePipelinesEnabled(pipelineEntity))) {
> ```

discussed offline to change condition
___
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 1036 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpsveL6Nin7A__KcPJ:-NjpsveL6Nin7A__KcPK:b-f5m5oq) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L1036)):*
> ```Java
>   @Owner(developers = RISHIKESH)
>   @Category(UnitTests.class)
>   public void testCheckIsSkipValidationForInlinePipelinesEnabled() throws IOException {
> ```

add one more UT having 2 conditions of setting returning true/false/exception
___
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 1040 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjptDCvD1WGI33gQ91Q:-NjptDCvD1WGI33gQ91R:b-y0h7m) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L1040)):*
> ```Java
>     SettingValueResponseDTO settingValueResponseDTO =
>         SettingValueResponseDTO.builder().value("true").valueType(SettingValueType.BOOLEAN).build();
>     doReturn(request)
> ```

request is not initiated

**Review by archit-harness — DISMISSED** [review]
> Reviewed 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)* status: 5 file(s) to review. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml` line 381 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-Njprxlr6lw4EaWYOnY4:-Nk-Gp9dBOEJSloUR-o6:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml#L381)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Skip Fail-Fast Validation Checks for Pipeline Execute
</blockquote></details>

done
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 458 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpsCpbBTm_Q6lbP8eq:-Nk-GtRd436UpVyd9XX7:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L458)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Keep it false
</blockquote></details>

done
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 468 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpsI4v6McP_GsnnCvO:-Nk-Gwh79xDjXoW8jEuN:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L468)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Convert string to Boolean and compare with Boolean.True.equals
</blockquote></details>

done
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 495 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpslZx5hTpVeJgKpEE:-Nk-GyiO-ci5CTfjcIC3:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L495)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

discussed offline to change condition
</blockquote></details>

done
___
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 1036 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjpsveL6Nin7A__KcPJ:-Nk-H1pa3809u5i3xJoO:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L1036)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

add one more UT having 2 conditions of setting returning true/false/exception
</blockquote></details>

done
___
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 1040 at r2](https://reviewable.io/reviews/harness/harness-core/55731#-NjptDCvD1WGI33gQ91Q:-Nk-H2LvAdyaIK5n0_n3:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/131d7ae2355eac74b983235eac3692e19e75e28c/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L1040)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

request is not initiated
</blockquote></details>

done

**Comment by yogesh-chauhan** [comment]
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 13 at r3](https://reviewable.io/reviews/harness/harness-core/55731#-NkEBWxKEjGkf_KV2XT6:-NkEBWxKEjGkf_KV2XT7:bxengit) ([raw file](https://github.com/harness/harness-core/blob/99b6f3cb697e416e3e4ef3246d15f0e091dd481b/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L13)):*
> ```Java
> import static io.harness.gitcaching.GitCachingConstants.BOOLEAN_FALSE_VALUE;
> import static io.harness.pms.contracts.plan.TriggerType.MANUAL;
> import static io.harness.rule.OwnerRule.*;
> ```

we should not do wild card imports. You will need to update your IDE settings to avoid this in future

**Comment by yogesh-chauhan** [comment]
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 19 at r3](https://reviewable.io/reviews/harness/harness-core/55731#-NkEBcDf2JqiYU5rVHId:-NkEBcDf2JqiYU5rVHIe:bxengit) ([raw file](https://github.com/harness/harness-core/blob/99b6f3cb697e416e3e4ef3246d15f0e091dd481b/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L19)):*
> ```Java
> import static org.mockito.ArgumentMatchers.*;
> import static org.mockito.ArgumentMatchers.eq;
> import static org.mockito.Mockito.*;
> ```

we should not do wild card imports. You will need to update your IDE settings to avoid this in future

**Comment by yogesh-chauhan** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 456 at r3](https://reviewable.io/reviews/harness/harness-core/55731#-NkEBjBR6m_nI5z3nD09:-NkEBjBR6m_nI5z3nD0A:bdimo1g) ([raw file](https://github.com/harness/harness-core/blob/99b6f3cb697e416e3e4ef3246d15f0e091dd481b/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L456)):*
> ```Java
>     return getPipelineYamlAndValidateStaticallyReferredEntities(pipelineJsonNode, pipelineEntity, start);
>   }
>   public boolean checkIsSkipFailFastValidationEnabled(PipelineEntity pipelineEntity) {
> ```

nit: we can omit the `check` prefix here. something like isSkipFailFastValidationEnabled

**Comment by yogesh-chauhan** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 467 at r3](https://reviewable.io/reviews/harness/harness-core/55731#-NkED-gn53qVP7tfJFIG:-NkED-gn53qVP7tfJFIH:b-9hnghq) ([raw file](https://github.com/harness/harness-core/blob/99b6f3cb697e416e3e4ef3246d15f0e091dd481b/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L467)):*
> ```Java
>       log.error("Failed to fetch setting value for {}", SKIP_FAIL_FAST_VALIDATION_CHECKS_FOR_PIPELINE_EXECUTE, ex);
>     }
>     return Boolean.valueOf(isSkipFailFastValidationEnabled).equals(Boolean.TRUE);
> ```

nit: we should always have constant on the left side of the operator like CONSTANT.equals(target). This is a good practise to follow in general to avoid NPEs

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 2 of 5 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)* status: 3 file(s) to review. 0 of 2 approvals obtained.

File: 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingIdentifiers.java (hunk 1 above)
Comment (rchaudharyy) [reply 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingIdentifiers.java]: done

**Comment by rchaudharyy** [comment]
trigger uts

**Review by rchaudharyy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)* status: 3 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 1034 at r1](https://reviewable.io/reviews/harness/harness-core/55731#gh-1400442401:-NkFk9kYBCH5i_duqx8n:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ae307be4da31f70dfa0adf1224598d3a175216e5/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L1034)):*
<details><summary><i>Previously, ankittiwari-harness (Ankit Tiwari) wrote…</i></summary><blockquote>

add test for case when setting is disabled.
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java` line 13 at r3](https://reviewable.io/reviews/harness/harness-core/55731#-NkEBWxKEjGkf_KV2XT6:-NkFkG4Z5mfv4_zeZdeM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/99b6f3cb697e416e3e4ef3246d15f0e091dd481b/pipeline-service/service/src/test/java/io/harness/pms/plan/execution/ExecutionHelperTest.java#L13)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

we should not do wild card imports. You will need to update your IDE settings to avoid this in future
</blockquote></details>

Done.

**Comment by rchaudharyy** [comment]
trigger ut5, ut6, ut7, ut9

**Comment by rchaudharyy** [comment]
trigger messagecheck

**Comment by rchaudharyy** [comment]
trigger ut5, ut7

**Comment by rchaudharyy** [comment]
trigger uts

**Comment by rchaudharyy** [comment]
trigger ut5, ut6, ut8, ut9

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 2 of 5 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55731)* status: 2. unresolved discussions.

**Comment by rchaudharyy** [comment]
trigger review

**Review by nishantsaini20 — COMMENTED** [review]
Approved for settings.

**Review by nishantsaini20 — APPROVED** [review]

**Pull request merged**